pkg/workflows/management/core_components.go

```diff
@@ -31,16 +31,15 @@ func (s *ensureEtcdCAPIComponentsExist) Checkpoint() *task.CompletedTask {
 	}
 }
 
-func (s *ensureEtcdCAPIComponentsExist) Restore(ctx context.Context, commandContext *task.CommandContext, completedTask *task.CompletedTask) (task.Task, error) {
+func (s *ensureEtcdCAPIComponentsExist) Restore(_ context.Context, _ *task.CommandContext, _ *task.CompletedTask) (task.Task, error) {
 	return &pauseGitOpsReconcile{}, nil
 }
 
 type upgradeCoreComponents struct {
 	UpgradeChangeDiff *types.ChangeDiff
 }
 
-// Run upgradeCoreComponents upgrades pre cluster upgrade components.
-func (s *upgradeCoreComponents) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+func upgradeCoreComponentsTask(ctx context.Context, commandContext *task.CommandContext) error {
```

Comment:
For what are you optimizing with this func? What value do we gain in maintainability or understandability by having it instead of putting its logic directly in `func (s *upgradeCoreComponents) Run`?

Reply:
As discussed with Guillermo, we want to reuse this functionality in the task of upgradeCoreComponentsMC, although it will introduce some code duplicity.

Reply:
This code can still be shared across funcs. `upgradeCoreComponentsMC` can create a `upgradeCoreComponents` struct and call its `Run` method. Im still having trouble understanding the value and for what is being optimized.

Reply:
Yeah, it's a little controversial, https://github.com/aws/eks-anywhere/pull/7238#discussion_r1439359609. Since I didn't have a strong opinion on it, I went to the approach proposed in that comment.

Reply:
i see. thanks. This is not clear from the code and will be lost in the code once this is merged. It also doesnt seem like the same thing here. I'm not talking about type asserting anything like this "`ntask.(*workflows.CollectMgmtClusterDiagnosticsTask)`"

I'm talking about reusing `upgradeCoreComponents`. something like:
```Go
upgradeCoreComponents{}.Run()
```